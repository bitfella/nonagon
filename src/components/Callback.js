import React from 'react';
import { Redirect } from 'react-router-dom';
import { appAccessToken } from 'config/spotify';
import { getHashKeyValue } from 'utils';
import Error from 'components/Error';
import Layout from 'components/Layout';

const Callback = () => {
  const accessToken = getHashKeyValue('access_token');

  if (accessToken) {
    localStorage.setItem(appAccessToken, accessToken);

    return <Redirect to="/" />;
  }

  return (
    <Layout>
      <div isLayoutBodyWrapper>
        <Error />
      </div>
    </Layout>
  );
};

export default Callback;
