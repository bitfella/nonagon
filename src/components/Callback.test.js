import React from 'react';
import { shallow } from 'enzyme';
import Callback from './Callback';

describe('Callback', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Callback />);

    expect(wrapper).toMatchSnapshot();
  });
});
