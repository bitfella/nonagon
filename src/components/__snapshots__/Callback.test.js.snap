// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callback renders correctly 1`] = `
<WithTheme(Layout)>
  <div
    isLayoutBodyWrapper={true}
  >
    <Error />
  </div>
</WithTheme(Layout)>
`;
